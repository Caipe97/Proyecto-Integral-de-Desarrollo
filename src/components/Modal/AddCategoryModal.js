import React, { useState } from 'react';
import Modal  from 'react-bootstrap/Modal';
import { Button } from 'react-bootstrap';
import TextField from '@material-ui/core/TextField';
import Add from '../../images/add.png';

const AddCategoryModal = (props) =>{
  const [modalShow, setModalShow] = useState(false);
  const [state, setState] = useState({
    name: '',
    errorMessage: ''
  })

  const handleChange = event => {
    setState({
      ...state,
      [event.target.name]: event.target.value
    })
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    await props.onCreateCategory(state.name, props.userId);
    setState({
      name: '',
      errorMessage: ''
    });
  };

  return (
    <>
      {/* <Button  type="button" variant="primary" onClick={() => setModalShow(true)}>
        Crear Categoría
      </Button> */}
      <img src={Add} alt='agregar' style={{width: '30px', height: '30px', cursor: 'pointer'}} onClick={() => setModalShow(true)}/>
      <Modal
        show={modalShow}
        size="lg"
        aria-labelledby="contained-modal-title-vcenter"
        centered
      >
        <Modal.Header>
          <Modal.Title id="contained-modal-title-vcenter">
          {'Crear Categoría'}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <form style={{ display: 'flex', justifyContent: 'center', alignItems: 'center', height: '100%', flexDirection: 'column' }}>
            <TextField label="Nombre" name='name' type='name' value={state.name} onChange={handleChange} required />
            <button onClick={handleSubmit} className='button'>Crear</button>
            <p>{state.errorMessage}</p>
          </form>
        </Modal.Body>
        <Modal.Footer>
          <Button type="button" onClick={() => setModalShow(false)}>Cerrar</Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
  
  export default AddCategoryModal;