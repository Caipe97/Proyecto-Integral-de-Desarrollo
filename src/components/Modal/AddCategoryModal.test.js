import React from 'react';
import { shallow } from 'enzyme'; 
import AddCategoryModal from './AddCategoryModal';
//import Combobox from "react-widgets/Combobox";
describe('ModalAddCategoryModal tests', () => {
   // const historyMock = { push: jest.fn(), };///
    //let mockProps;
    const preventDefault = { preventDefault: jest.fn() }
    let wrapper;
    beforeEach(() => {
        let mockProps= {
            state:{
                name: '',
                errorMessage: ''
            },
            preventDefault:jest.fn(),
            onCreateCategory:jest.fn()
        };
        wrapper = shallow(<AddCategoryModal {...mockProps} />
            
        );
        wrapper.find('[name="name"]').at(0).simulate('change', { target: { name: 'name', value: 'abc' } });

    })
    it('expect to render AddCategoryModal component to render', () => {
        expect(wrapper).toMatchSnapshot();
 //       expect(wrapper.find('calculateTotalCaloriesPerMeal')).toBeDefined();
    })
    it('Set modal and setstate null message expect to change  onclick buttons',  () => {
       
    wrapper.find('[type="button"]').at(0).simulate('click');
       // await wrapper.find('[type="button"]').at(2).simulate('click');
       expect(wrapper.find('div')).toBeDefined();
        expect(wrapper.find('div.p').contains('')).toBeDefined();

      
    })
    it('expect to change  onclick buttons for set modal', () => {
       
        wrapper.find('img').at(0).simulate('click');
               
    })

    it('handleSubmit expect to change onclick for crear', () => {
        ///jest.useFakeTimers();

         wrapper.find('[className="button"]').at(0).simulate('click', preventDefault);


    })


})
