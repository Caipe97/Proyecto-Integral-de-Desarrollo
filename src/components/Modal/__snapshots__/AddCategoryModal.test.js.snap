// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalAddCategoryModal tests expect to render AddCategoryModal component to render 1`] = `
<Fragment>
  <img
    alt="agregar"
    onClick={[Function]}
    src="add.png"
    style={
      Object {
        "cursor": "pointer",
        "height": "30px",
        "width": "30px",
      }
    }
  />
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    restoreFocus={true}
    show={false}
    size="lg"
  >
    <ModalHeader
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Crear Categor√≠a
      </ModalTitle>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "white",
            "color": "black",
          }
        }
        type="button"
        variant="primary"
      >
        X
      </Button>
    </ModalHeader>
    <ModalBody>
      <form
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "justifyContent": "center",
          }
        }
      >
        <WithStyles(ForwardRef(TextField))
          label="Nombre"
          name="name"
          onChange={[Function]}
          required={true}
          type="name"
          value="abc"
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Crear
        </button>
        <p />
      </form>
    </ModalBody>
  </Modal>
</Fragment>
`;
